cmd/cli/kubectl-kyverno/commands/test/test.go

```diff
@@ -98,6 +98,51 @@ func runTest(out io.Writer, testCase test.TestCase, registryAccess bool, auditWa
 	} else {
 		fmt.Fprintln(out, "  Applying", len(policies)+len(validatingAdmissionPolicies), pluralize.Pluralize(len(policies)+len(validatingAdmissionPolicies), "policy", "policies"), "to", len(uniques), pluralize.Pluralize(len(uniques), "resource", "resources"), "...")
 	}
+	rulesmap := make(map[string]string)
+	policymap := make(map[string]string)
+	for _, res := range testCase.Test.Results {
+		if res.IsValidatingAdmissionPolicy {
+			continue
+		}
+		var ns string
+		var n bool
+		r := false
+		for _, policy := range policies {
+			if policy.IsNamespaced() {
+				ns = policy.GetNamespace()
+			}
+			if res.Policy == policy.GetName() || res.Policy == "default/"+policy.GetName() || res.Policy == ns+"/"+policy.GetName() {
+				n = true
+				for _, rule := range autogen.ComputeRules(policy) {
+					if res.Rule == rule.Name {
+						r = true
+						continue
+					}
+				}
+			}
+		}
+		if !r {
+			rulesmap[res.Policy] = res.Rule
+		}
+		if !n {
+			policymap[res.Policy] = res.Policy
+		}
```

Comment:
There is no need for these conditions. Just store them directly in the loop.

Reply:
What if 2 rules are present and first one mismatches then ,if we store them directly on loop then it errors like rule not found without checking the second one.
So we first check the all rules name and if any one matches with result rule name then we make it true else false . Then we add the rule in map , if it is not true  i.e rule not found